File: api_call/send_audio.js
const fs = require("fs");
const path = require("path");
const axios = require("axios");
const FormData = require("form-data");

/**
 * Upload an audio file to the Flask API
 *
 * @param {string} apiUrl - The URL of the upload endpoint
 * @param {string} filePath - Path to the audio file to upload
 * @returns {Promise<object>} - Response from the server
 */
async function uploadAudioFile(apiUrl, filePath) {
  // Check if file exists
  if (!fs.existsSync(filePath)) {
    throw new Error(`Audio file not found: ${filePath}`);
  }

  // Check if file is an MP3
  if (path.extname(filePath).toLowerCase() !== ".mp3") {
    throw new Error("File must be an MP3");
  }

  try {
    // Create a FormData instance and append the file
    const formData = new FormData();
    formData.append("audio", fs.createReadStream(filePath), {
      filename: path.basename(filePath),
      contentType: "audio/mpeg",
    });

    // Make the POST request
    const response = await axios.post(apiUrl, formData, {
      headers: {
        ...formData.getHeaders(),
      },
    });

    // Return the response data
    return response.data;
  } catch (error) {
    console.error(`Error uploading file: ${error.message}`);
    return null;
  }
}

// Example usage
(async () => {
  // URL of your Flask API endpoint
  const API_URL = "https://940c-35-221-203-99.ngrok-free.app/api/upload/audio";

  // Path to your MP3 file
  const AUDIO_FILE_PATH = "message1.mp3"; // Replace with your actual file path

  // Number of parallel API calls
  const numCalls = 2;

  try {
    // Create an array of promises for parallel API calls
    const promises = Array.from({ length: numCalls }, () =>
      uploadAudioFile(API_URL, AUDIO_FILE_PATH)
    );

    // Wait for all promises to resolve
    const results = await Promise.all(promises);

    results.forEach((result, index) => {
      if (result) {
        console.log(`Upload ${index + 1} successful!`);
        console.log("Server response:", result);
        console.log(`File can be accessed at: ${result.path}`);
      } else {
        console.log(`Upload ${index + 1} failed!`);
      }
    });
  } catch (error) {
    console.error(`Error: ${error.message}`);
  }
})();

File: api_call/send_audio.py
import requests
import os

def upload_audio_file(api_url, file_path):
    """
    Upload an audio file to the Flask API
    
    Args:
        api_url (str): The URL of the upload endpoint
        file_path (str): Path to the audio file to upload
    
    Returns:
        dict: Response from the server
    """
    # Check if file exists
    if not os.path.exists(file_path):
        raise FileNotFoundError(f"Audio file not found: {file_path}")
    
    # Check if file is mp3
    if not file_path.lower().endswith('.mp3'):
        raise ValueError("File must be an MP3")
    
    try:
        # Open the file in binary mode
        with open(file_path, 'rb') as audio_file:
            # Create the files dictionary for the request
            files = {
                'audio': (os.path.basename(file_path), audio_file, 'audio/mpeg')
            }
            
            # Make the POST request
            response = requests.post(api_url, files=files)
            
            # Check if request was successful
            response.raise_for_status()
            
            # Return the response data
            return response.json()
            
    except requests.exceptions.RequestException as e:
        print(f"Error uploading file: {e}")
        return None

# Example usage
if __name__ == "__main__":
    # URL of your Flask API endpoint
    API_URL = "https://940c-35-221-203-99.ngrok-free.app/api/upload/audio"
    
    # Path to your MP3 file
    AUDIO_FILE_PATH = "full.mp3"  # Replace with your actual file path
    
    try:
        # Upload the file
        result = upload_audio_file(API_URL, AUDIO_FILE_PATH)
        
        if result:
            print("Upload successful!")
            print("Server response:", result)
            print(f"File can be accessed at: {result.get('path')}")
        else:
            print("Upload failed!")
            
    except FileNotFoundError as e:
        print(f"Error: {e}")
    except ValueError as e:
        print(f"Error: {e}")
    except Exception as e:
        print(f"Unexpected error: {e}")